lib/datadog/tracing/contrib/aws/services/dynamodb.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+require_relative '../ext'
+
+def add_dynamodb_tags(span, params)
```

Comment:
Each of these service-specific files would look like:
```ruby
module Datadog
  module Tracing
    module Contrib
      module Aws
        module Service
          class DynamoDB < Base
            def add_tags(span, params)
              table_name = params[:table_name]
              span.set_tag(Aws::Ext::TAG_TABLE_NAME, table_name)
            end
          end
        end 
      end 
    end 
  end 
end
```

You'd also create a base implemented, just in case case we want a fallback behaviour (in the `base.rb` file):
```ruby
module Datadog
  module Tracing
    module Contrib
      module Aws
        module Service
          class Base
            def add_tags(span, params)
            end
          end
        end
      end
    end
  end
end
```